Content.Server/Remotes/DoorRemoteSystem.cs

```diff
@@ -0,0 +1,41 @@
+using Robust.Shared.GameObjects;
+using Robust.Shared.Localization;
+using Robust.Shared.Player;
+using Content.Shared.Interaction;
+using Content.Shared.Popups;
+
+namespace Content.Server.Remotes
+{
+    public sealed class DoorRemoteSystem : EntitySystem
+    {
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<DoorRemoteComponent, UseInHandEvent>(OnInHandActivation);
+        }
+
+        public void OnInHandActivation(EntityUid user, DoorRemoteComponent component, UseInHandEvent args)
+        {
+            var popupSystem = EntityManager.EntitySysManager.GetEntitySystem<SharedPopupSystem>();
```

Comment:
This applies to this and also the code in airlock system where you get `SharedInteractionSystem`

Instead of getting the system inside of this function, what you should do is add a 
```cs
[Dependency] private readonly SharedPopupSystem _popupSystem = default!;
```
to the beginning of the this class, and then just use `_popupSystem` in the function.

The difference is just that this system will only get and cache the popup-system once, while the current code will retrieve the system every time a door component is used. It's just slightly faster. Not super important for door remotes I guess, but just a good practice to follow.

Unfortunately the whole system-dependency thing only works inside entity systems so you will still see EntitySystem.Get<>() calls used outside of systems (e.g., in components), but thats usually old code that should really be moved into a system

Reply:
System dependencies added.